examples/basics/dataframe_usage.py

```diff
@@ -0,0 +1,171 @@
+"""
+Basic DataFrame Usage Examples
+
+This example demonstrates how flyte.DataFrame is a wrapper around raw dataframes.
+
+Key principles:
+- Working with raw dataframes (pd.DataFrame) = automatic upload/download
+- Working with flyte.DataFrame = I/O only when you explicitly request it
+- At task completion: all dataframes are uploaded to remote storage
+"""
+
+import pandas as pd
+
+import flyte
+from flyte.io import DataFrame
+
+# Create task environment with required dependencies
+img = flyte.Image.from_debian_base()
+img = img.with_pip_packages("pandas", "pyarrow")
+
+env = flyte.TaskEnvironment(
+    "dataframe_usage",
+    image=img,
+    resources=flyte.Resources(cpu="1", memory="2Gi"),
+)
+
+# Simple sample data
+SAMPLE_DATA = {"name": ["Alice", "Bob", "Charlie"], "age": [25, 30, 35], "city": ["NYC", "SF", "LA"]}
+
+
+@env.task
+async def create_raw_dataframe() -> pd.DataFrame:
+    """
+    Task returns RAW dataframe (pd.DataFrame).
+    UPLOAD happens at the end of the task, the Flyte Type Engine will do the upload from memory to blob store.
+    """
+    return pd.DataFrame(SAMPLE_DATA)
+
+
+@env.task
+async def create_wrapper_dataframe() -> DataFrame:
```

Comment:
i want ot stick to using flyte.io.DataFrame, that way it can be understood quickly